src/MerchantCenter/PolicyComplianceCheck.php

```diff
@@ -0,0 +1,240 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\WC;
+use Automattic\WooCommerce\GoogleListingsAndAds\Google\GoogleHelper;
+use Automattic\WooCommerce\GoogleListingsAndAds\PluginHelper;
+use Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter\TargetAudience;
+
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class PolicyComplianceCheck
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter
+ *
+ * @since x.x.x
+ */
+class PolicyComplianceCheck implements Service {
+
+	use PluginHelper;
+
+	/**
+	 * The WC proxy object.
+	 *
+	 * @var wc
+	 */
+	protected $wc;
+
+	/**
+	 * @var GoogleHelper
+	 */
+	protected $google_helper;
+
+	/**
+	 * @var TargetAudience
+	 */
+	protected $target_audience;
+
+	/**
+	 * BaseController constructor.
+	 *
+	 * @param WC             $wc
+	 * @param GoogleHelper   $google_helper
+	 * @param TargetAudience $target_audience
+	 */
+	public function __construct( WC $wc, GoogleHelper $google_helper, TargetAudience $target_audience ) {
+		$this->wc              = $wc;
+		$this->google_helper   = $google_helper;
+		$this->target_audience = $target_audience;
+	}
+
+	/**
+	 * Check if the store website is accessed by all users for the controller.
+	 *
+	 * @return bool
+	 */
+	public function is_accessible(): bool {
+		$all_allowed_countries = $this->wc->get_allowed_countries();
+		$target_countries      = $this->target_audience->get_target_countries();
+
+		foreach ( $target_countries as $country ) {
+			if ( ! array_key_exists( $country, $all_allowed_countries ) ) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Check if the store sample product landing pages lead to a 404 error.
+	 *
+	 * @return bool
+	 */
+	public function has_page_not_found_error(): bool {
+		$headers = $this->get_landing_page_headers();
+		if ( ! empty( $headers ) && $headers[0] !== 'HTTP/1.1 200 OK' ) {
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Check if the store sample product landing pages has redirects through 3P domains.
+	 *
+	 * @return bool
+	 */
+	public function has_redirects(): bool {
+				$headers = $this->get_landing_page_headers();
```

Comment:
Even though the PHPCS config of this repo doesn't check indentations with the `exact` ruleset, it would be great if the indentations could be aligned here.

Reply:
Done.